api/comment_threads.rb

```diff
@@ -27,9 +27,12 @@
     error 404, [t(:requested_object_not_found)].to_json
   end
 
-  if params["user_id"] and bool_mark_as_read
+  if params["user_id"]
     user = User.only([:id, :username, :read_states]).find_by(external_id: params["user_id"])
-    user.mark_as_read(thread) if user
+  end
+
+  if user and bool_mark_as_read
+    user.mark_as_read(thread)
   end
```

Comment:
@jcdyer user being null does not set user related fields on returned Thread e.q. `read` status. If request has `user_id` (which it should always have in my opinion) then we get user and use it to prepare thread.
`bool_mark_as_read` should not be there at all; as we are no longer marking thread as read on GET endpoint for mobile APIs, but Forum has other clients too (Web app) and I tried to keep default behaviour as is and I also believe this is making Forums app messy everyday.
If `user` is never assigned; it negates the `if` case and does not throw any exception; works fine for ruby.
"Can a student mark a thread as already having been read by another student?" @nasthagiri if i am not wrong then yes, we keep track of read status for multiple users against a thread. right?